src/hooks/http.ts

```diff
@@ -48,15 +48,27 @@ export class Http {
       GlobalDurationTimer.start();
       if (isEmptyString(requestData.body)) {
         const body = extractBodyFromWriteOrEndFunc(args);
-        if (body) {
-          requestData.body += body.substr(0, getEventEntitySize(false));
-        }
-        if (currentSpan) currentSpan.info.httpInfo = getHttpInfo(requestData, {});
+        Http.aggregateRequestBodyToSpan(body, requestData, currentSpan, getEventEntitySize(true));
       }
       GlobalDurationTimer.stop();
     };
   }
 
+  static aggregateRequestBodyToSpan(
+    body,
+    requestData,
+    currentSpan,
+    maxSize = getEventEntitySize(true)
+  ) {
+    if (body && !requestData.truncated) {
+      requestData.body += body;
+      const truncated = maxSize < requestData.body.length;
+      if (truncated) requestData.body = requestData.body.substr(0, maxSize);
+      requestData.truncated = truncated;
+    }
+    if (currentSpan) currentSpan.info.httpInfo = getHttpInfo(requestData, {});
```

Comment:
Do we need to do `getHttpInfo` before the truncate?

Reply:
it was always outside the if
if (body && !requestData.truncated) {